### ISSUE TITLE
yoda should treat negative numbers as literals

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1571]
Even though [negative numbers aren't literals](http://esprima.org/demo/parse.html?code=-1), I think they should be treated as if they are.

``` js
// test.js
if (1 < x) {}
if (-1 < x) {}
if (x < 1) {}
if (x < -1) {}
```

```
$ eslint --reset --no-eslintrc --rule "yoda: 2" test.js

test.js
  2:4  error  Expected literal to be on the right side of <  yoda

✖ 1 problem
```

To me, that should have two errors.
